### DISCUSSION TITLE
Is wlvncc client compatible with tigervncserver?

### DISCUSSION THREAD

**Discussion by bob333** [any1/wlvncc#28] [General]
Hi,
    Can this client connect to tigervncserver ?

**Comment by any1** [comment]
Probably. Why don't you just try it?

**Reply by bob333** [reply]
I did. It failed to connect to server with the following error message. Therefore, I want to confirm before investing significant time into examining the source code. Additionally, could you please clarify how to pass the credentials of the vnc server to wlvncc? Should they be entered through the command line or does the program search for a configuration file in a predetermined location? 

$ ./build/wlvncc my-remote-server 5901
Using EGL for rendering...
17/04/2023 00:56:22 VNC server supports protocol version 3.8 (viewer 3.8)
17/04/2023 00:56:22 We have 1 security types to read
17/04/2023 00:56:22 0) Received security type 19
17/04/2023 00:56:22 Selecting security type 19 (0/1 in the list)
17/04/2023 00:56:22 Selected Security Scheme 19
17/04/2023 00:56:22 GnuTLS version 3.7.3 initialized.
17/04/2023 00:56:22 Got VeNCrypt version 0.2 from server.
17/04/2023 00:56:22 We have 1 security types to read
17/04/2023 00:56:22 0) Received security type 259
17/04/2023 00:56:22 Selecting security type 259 (0/1 in the list)
17/04/2023 00:56:22 TLS session initialized.
17/04/2023 00:56:22 TLS anonymous credential created.
17/04/2023 00:56:22 TLS handshake blocking.
17/04/2023 00:56:23 TLS handshake blocking.
17/04/2023 00:56:24 TLS handshake done.
17/04/2023 00:56:24 GetCredential callback is not set.
Failed to connect to server
Exiting...

**Reply by any1** [reply]
Yeah, the server is trying to ask for a password and/or ssl certificates but this isn't implemented in wlvncc.

The recommended approach for now is to use ssh forwarding or wireguard instead of TLS.

Although, implementing that callback probably isn't very hard in case you want to give it a go.

**Reply by bob333** [reply]
To clarify, would this setting be configured on the server side rather than just on the client side? I ask because I don't have authority over the server.

**Reply by any1** [reply]
Yes, the authentication scheme is controlled by the server.

I use this project mostly to test and benchmark wayvnc. It also serves as a tech demo for others to imitate in some cases (such as end-to end hardware accelerated h264).

You might be better off just running TigerVNC' vncviewer.

**Reply by bob333** [reply]
Thanks for confirming.

**Reply by dezeroku** [reply]
FWIW, I can confirm that current master (`2b9a886edd38204ef36e9f9f65dd32aaa3784530`) works fine with password-protected tiger`vncserver` without TLS, tunneled over ssh.
